mmv1/templates/terraform/examples/region_url_map_default_route_action.tf.erb

```diff
@@ -0,0 +1,101 @@
+resource "google_compute_region_url_map" "<%= ctx[:primary_resource_id] %>" {
+  region = "us-central1"
+
+  name        = "<%= ctx[:vars]['region_url_map_name'] %>"
+  description = "a description"
+
+  default_route_action {
+    retry_policy {
+      retry_conditions = [
+        "5xx",
+        "gateway-error",
+      ]
+      num_retries = 3
+      per_try_timeout {
+        seconds = 0
+        nanos = 500
+      }
+    }
+    request_mirror_policy {
+      backend_service = google_compute_region_backend_service.home.id
+    }
+    weighted_backend_services {
```

Comment:
Can you add another weighted_backend_service here so we test at least 2 items in the array?

Part of me wonders if this will be reordered by the API or something like that, which could cause problems for users if we don't mark it as a set from the start

Reply:
Yeah, good call!

Reply:
@slevenick I added a new `weighted_backend_services` block in 2005910 (and extra blocks for other nested arrays like `weighted_backend_services[0].header_action.request_headers_to_add`) and the tests don't show any un-empty plans after a refresh step so it seems like the API doesn't reorder lists.

**However, should I swap to using a Set now anyway?** I'm aware there are some bugs around Set behaviour so I wasn't sure what the balance of pros/cons is. I investigated and the only immediate downside of using a Set for `weighted_backend_services` was that the value for `weighted_backend_services.backend_service` would always need to be a `self_link` instead of an `id`

Reply:
I just pushed a commit [2bcafac](https://github.com/GoogleCloudPlatform/magic-modules/pull/6674/commits/2bcafacba426bd2116eb920a2e198e4386db7da0) exploring this, but then realised that I'm making a regional version of a resource that already has a global version where all these fields exist. I'll need to make sure they match up otherwise users will get confused!

For example, `weighted_backend_services.backend_service` isn't a set in `google_compute_url_map` : https://github.com/hashicorp/terraform-provider-google/blob/51686cec843e2c9f685b83b183ab1d722b3fda17/google/resource_compute_url_map.go#L386-L397

I'll identify any diffs between this PR and that resource and fix this PR to match

Reply:
After comparing the two resources I've reverted the work to make fields in this PR sets, as I think the two resources should work the same way

| RegionUrlMap | UrlMap |
|---|---|
| weightedBackendServices added in this PR | [weightedBackendServices](https://github.com/hashicorp/magic-modules/blob/4f1ef3974f99d6a9efa95ea0284ee831cc63d2f5/mmv1/products/compute/api.yaml#L18566-L18670) |
| retryPolicy added in this PR | [retryPolicy](https://github.com/hashicorp/magic-modules/blob/4f1ef3974f99d6a9efa95ea0284ee831cc63d2f5/mmv1/products/compute/api.yaml#L18736-L18809) |
| requestMirrorPolicy  added in this PR | [requestMirrorPolicy](https://github.com/hashicorp/magic-modules/blob/4f1ef3974f99d6a9efa95ea0284ee831cc63d2f5/mmv1/products/compute/api.yaml#L18810-L18831) |